linkerd/tls/src/imp/openssl.rs

```diff
@@ -0,0 +1,295 @@
+use crate::{ClientId, HasNegotiatedProtocol, NegotiatedProtocolRef};
+use linkerd_identity::{ClientConfig, Name, ServerConfig};
+use linkerd_io::{self as io, AsyncRead, AsyncWrite, ErrorKind, PeerAddr, ReadBuf, Result};
+use std::net::SocketAddr;
+use std::{
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use std::str::FromStr;
+use tracing::debug;
+
+#[cfg(feature = "boring-tls")]
+use {
+    boring::{
+        fips, ssl,
+        ssl::{SslAcceptor, SslConnector, SslConnectorBuilder, SslMethod, SslVerifyMode},
+        version,
+        x509::store::X509StoreBuilder,
+    },
+    tokio_boring::SslStream,
+};
+#[cfg(not(feature = "boring-tls"))]
+use {
+    openssl::{
+        fips, ssl,
+        ssl::{Ssl, SslAcceptor, SslConnector, SslConnectorBuilder, SslMethod, SslVerifyMode},
+        version,
+        x509::store::X509StoreBuilder,
+    },
+    tokio_openssl::SslStream,
+};
+
+#[derive(Clone)]
+pub struct TlsConnector(ssl::SslConnector);
+
+impl TlsConnector {
+    #[cfg(not(feature = "boring-tls"))]
+    pub async fn connect<IO>(&self, domain: Name, stream: IO) -> Result<client::TlsStream<IO>>
+    where
+        IO: AsyncRead + AsyncWrite + Unpin,
+    {
+        let ssl = self
+            .0
+            .configure()
+            .unwrap()
+            .into_ssl(domain.as_ref())
+            .unwrap();
+        let mut s = TlsStream::new(ssl, stream);
+        match Pin::new(&mut s.0).connect().await {
+            Ok(_) => Ok(s),
+            Err(err) => Err(io::Error::new(ErrorKind::Other, err)),
+        }
+    }
+    #[cfg(feature = "boring-tls")]
+    pub async fn connect<IO>(&self, domain: Name, stream: IO) -> Result<client::TlsStream<IO>>
+    where
+        IO: AsyncRead + AsyncWrite + Unpin,
+    {
+        let conf = self.0.configure().unwrap();
+        match tokio_boring::connect(conf, domain.as_ref(), stream).await {
+            Ok(ss) => Ok(ss.into()),
+            Err(err) => {
+                let _ = err.ssl();
+                Err(io::Error::new(ErrorKind::Other, "Ble"))
+            }
+        }
+    }
+}
+
+impl From<SslConnector> for TlsConnector {
+    fn from(connector: SslConnector) -> Self {
+        Self(connector)
+    }
+}
+
+impl From<SslConnectorBuilder> for TlsConnector {
+    fn from(builder: SslConnectorBuilder) -> Self {
+        builder.build().into()
+    }
+}
+
+impl From<Arc<ClientConfig>> for TlsConnector {
+    fn from(conf: Arc<ClientConfig>) -> Self {
+        println!("Setting up ssl connector");
```

Comment:
let's either remove the `println!` or replace it with a `debug!` or something, please :)